🌐 Update Urdu translation for `docs/ur/docs/benchmarks.md`

AhsanSheraz opened the pull request.

AhsanSheraz committed:
update: add translation of external-links.md

AhsanSheraz committed:
fix: update location of external-links file

tiangolo commented:
📝 Docs preview for commit b0efbaaceb13c1da34283977d3c9d5959f784880 at: https://3fc208db.fastapitiangolo.pages.dev

AhsanSheraz committed:
fix: links and english words

tiangolo commented:
📝 Docs preview for commit 641e549d254b821d13938baeec4e7b5822aa70b4 at: https://8aa88612.fastapitiangolo.pages.dev

AhsanSheraz committed:
fix: articles links

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

AhsanSheraz committed:
fix: spaces between headings

AhsanSheraz committed:
Merge branch 'add-urdu-translation-external-link-page' of https://github.com/AhsanSheraz/fastapi into add-urdu-translation-external-link-page

talhaumer commented:
Looks Good

AhsanSheraz commented:
@tiangolo  Can you please add labels to this MR?

talhaumer reviewed (approved):
Looks good

talhaumer commented:
Looks good

talhaumer reviewed (approved):
Looks good

talha-softversum reviewed (approved):
Looks Good

tiangolo commented on docs/ur/docs/benchmarks.md:
```diff
@@ -1,4 +1,4 @@
-# بینچ مارکس
+# <div style="text-align: right;">بینچ مارکس</div>
```
Is this necessary? 🤔 

I would think maybe there's then a configuration missing in MkDocs... I'll have to investigate it.

AhsanSheraz replied:
@tiangolo previously it was left aligned so I have to add this you can check and let me know. Also I simply inherit the parent (en) MkDocs in my ur folder MkDocs file.

talhaumer reviewed (approved):
Looking Good

tiangolo committed:
Merge branch 'master' into add-urdu-translation-external-link-page

tiangolo reviewed (changes requested):
Hello! I discovered there's was a bug in how I set up MkDocs Material, that's why all the CSS styles to set up text right-to-left were needed. Now that it's fixed in https://github.com/tiangolo/fastapi/pull/10490 none of that should be needed.

I also updated how this page is structured here: https://github.com/tiangolo/fastapi/pull/10495

So now the content to translate should be smaller, e.g. no translations needed for each language name, as all that is taken from the config. In that same PR I also updated the existing languages, you could do the same update here. 🙏 

Thanks! ☕

AhsanSheraz committed:
fix: translation right to left div

AhsanSheraz committed:
Merge branch 'master' into add-urdu-translation-external-link-page

AhsanSheraz committed:
update: external link

AhsanSheraz committed:
Merge branch 'add-urdu-translation-external-link-page' of https://github.com/AhsanSheraz/fastapi into add-urdu-translation-external-link-page

tiangolo commented:
📝 Docs preview for commit 143104b927476065471b3005920563fe8a66f63b at: https://839e56a4.fastapitiangolo.pages.dev

AhsanSheraz commented:
> Hello! I discovered there's was a bug in how I set up MkDocs Material, that's why all the CSS styles to set up text right-to-left were needed. Now that it's fixed in #10490 none of that should be needed.
> 
> I also updated how this page is structured here: #10495
> 
> So now the content to translate should be smaller, e.g. no translations needed for each language name, as all that is taken from the config. In that same PR I also updated the existing languages, you could do the same update here. 🙏
> 
> Thanks! ☕

@tiangolo fixed the things that you told.

talhaumer reviewed (approved):
Looking Good

tiangolo reviewed (changes requested):
Thanks @AhsanSheraz! 

Now that the RTL bug is fixed, please remove all the extra `style="direction: rtl;"` and their divs. Thanks! ☕

AhsanSheraz committed:
update: remove rtl divs

AhsanSheraz committed:
Merge branch 'master' into add-urdu-translation-external-link-page

AhsanSheraz commented:
> Thanks @AhsanSheraz!
> 
> Now that the RTL bug is fixed, please remove all the extra `style="direction: rtl;"` and their divs. Thanks! ☕

Done @tiangolo

tiangolo commented:
📝 Docs preview for commit 6929a7069762762276c569b2c28650df80c236d1 at: https://86521267.fastapitiangolo.pages.dev

AhsanSheraz commented:
@tiangolo everything is working good but we have to add this <div style="text-align: right;"> because it is making align to left but it should be at right, so adding this as well.
![image](https://github.com/tiangolo/fastapi/assets/51913596/9476584a-0e07-49d9-96e6-67722ace9e39)

AhsanSheraz committed:
update: fix make it align right

tiangolo commented:
📝 Docs preview for commit 2b1c77485bc98c5ae2a443f6271e17eb84e0c741 at: https://5ff66add.fastapitiangolo.pages.dev

tiangolo commented:
I just noticed that there was another CSS bug introduced by a previous RTL PR, I just fixed it here: https://github.com/tiangolo/fastapi/pull/11039

Now with this, you can remove all the `style="text-align: right;"` blocks. 🤓

tiangolo committed:
Merge branch 'master' into add-urdu-translation-external-link-page

tiangolo commented:
📝 Docs preview for commit 6cebf0599cc2daf70d99e96bb016b3f6c2b26c27 at: https://05a3595f.fastapitiangolo.pages.dev

AhsanSheraz committed:
update: fix remove style text-align right

AhsanSheraz committed:
Merge branch 'master' into add-urdu-translation-external-link-page

tiangolo commented:
📝 Docs preview for commit 82a9dd4592d029860adbb3ca7d59a6c6a5e12626 at: https://1e425d33.fastapitiangolo.pages.dev

AhsanSheraz committed:
removed: all div that added for styles previously

tiangolo commented:
📝 Docs preview for commit 6c9362e268ff48273b53f379fc56323f05ba2d29 at: https://28dd6e57.fastapitiangolo.pages.dev

AhsanSheraz reviewed (commented):
Fixed all the changes in this [MR](https://github.com/tiangolo/fastapi/pull/10046)

AhsanSheraz commented:
> I just noticed that there was another CSS bug introduced by a previous RTL PR, I just fixed it here: #11039
> 
> Now with this, you can remove all the `style="text-align: right;"` blocks. 🤓

@tiangolo Fixed all the changes in this MR related to `style="text-align: right;" divs and removed this from all the files.

aimasheraz1 reviewed (approved):
Looks good now without style and div

tiangolo committed:
🔥 Remove no longer translated file

tiangolo committed:
Merge branch 'master' into add-urdu-translation-external-link-page

tiangolo commented:
Thank you! :nerd_face: 

We are no longer translating some specific pages: https://fastapi.tiangolo.com/contributing/#dont-translate-these-pages

So I updated it to include the updated page that would go in. Thanks! :cake:
